packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -88,14 +93,45 @@ type CreateSearchIndexCancelledAction = {
   type: ActionTypes.CreateSearchIndexCancelled;
 };
 
+type OpenUpdateSearchIndexModalAction = {
+  type: ActionTypes.OpenUpdateSearchIndexModal;
+  indexName: string;
+  indexDefinition: string;
+};
+
+type UpdateSearchIndexStartedAction = {
+  type: ActionTypes.UpdateSearchIndexStarted;
+};
+
+type UpdateSearchIndexFailedAction = {
+  type: ActionTypes.UpdateSearchIndexFailed;
+  error: string;
+};
+
+type UpdateSearchIndexSucceededAction = {
+  type: ActionTypes.UpdateSearchIndexSucceeded;
+};
+
+type UpdateSearchIndexCancelledAction = {
+  type: ActionTypes.UpdateSearchIndexCancelled;
+};
+
 type CreateSearchIndexState = {
   isModalOpen: boolean;
   isBusy: boolean;
 };
 
+type UpdateSearchIndexState = {
+  isModalOpen: boolean;
+  isBusy: boolean;
+  indexName: string;
+  indexDefinition: string;
+};
```

Comment:
```suggestion
type UpdateSearchIndexState = {
  isModalOpen: false;
} | {
  isModalOpen: true;
  isBusy: boolean;
  indexName: string;
};
```

This way we do not have to set the indexName as default. We can just set it as `updateIndex: { isModalOpen: false }`

Reply:
If we do this, the spread operator won't work in the reducer, as it can not infer the current type (isModalOpen can change on runtime), so we will need to specify every field one by one.

Reply:
don't have a strong opinion here, but if we leave it as it, then we do not need to have the default value for the indexName. technically it can be a null or a string.